### PR TITLE
Don't autocomplete inside quotes

### PR DESCRIPTION
**Pull request by proteusvacuum**

### CODE REVIEW

File: corehq/apps/reports/static/reports/js/filters/case_list_explorer_knockout_bindings.js [hunk 1]
```diff
@@ -51,10 +51,17 @@ hqDefine("reports/js/filters/case_list_explorer_knockout_bindings", ['jquery', '
...
         },
         update: function(element, valueAccessor){
+            var caseProperties = ko.utils.unwrapObservable(valueAccessor());
             var casePropertyAutocomplete = {
                 getCompletions: function(editor, session, pos, prefix, callback){
-                    var data = ko.utils.unwrapObservable(valueAccessor());
-                    callback(null, _.map(data, function(suggestion){
+                    var currentValue = editor.getValue(),
+                        leftQuotes = (currentValue.substr(0, pos.column).match(/'|"/g) || []).length,
+                        insideQuote =  leftQuotes && (leftQuotes % 2 !== 0);
```
Comment (pr33thi) [inline corehq/apps/reports/static/reports/js/filters/case_list_explorer_knockout_bindings.js]: Can you have escaped quotes in a case property? That's the only edge case I can think of that is maybe not handled here

Comment (proteusvacuum) [reply corehq/apps/reports/static/reports/js/filters/case_list_explorer_knockout_bindings.js]: probably. this also doesn't check if you have something like "hello 'pr33thi'" which I can see being something someone would do. Will update.

**Comment by orangejenny** [comment]
Not sure how deeply you want to go down this hole, but the general problem of figuring out whether or not you're inside quotes gets pretty complicated and I don't think is doable in a regex (at least not a readable one). App manager [does it like this](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/app_manager/xpath.py#L35-L70).

**Comment by orangejenny** [comment]
Nevermind, ignore that, since you don't have to find & replace this should be doable.

**Comment by proteusvacuum** [comment]
new buddy @calellowitz

**Comment by proteusvacuum** [comment]
:white_check_mark:

**Review by pr33thi — APPROVED** [review]

**Pull request merged**